lib/Controller/MfazonesController.php

```diff
@@ -59,18 +63,32 @@ public function __construct(
         $this->groupManager = $groupManager;
         $this->tagManager = $tagManager;
         $this->tagMapper = $tagMapper;
+		$this->session = $session;
         $this->systemTagManager = $systemTagManager;
     }
 
     public function hasAccess($source)
     {
         try {
+            $mfaVerified = '0';
+            if (!empty($this->session->get('globalScale.userData'))) {
+                $attr = $this->session->get('globalScale.userData')["userData"];
+                $mfaVerified = $attr["mfaVerified"];
+            }
+            if (!empty($this->session->get('user_saml.samlUserData'))) {
+                $attr = $this->session->get('user_saml.samlUserData');
+                $mfaVerified = $attr["mfa_verified"][0];
+            }
+            if (!empty($this->session->get("two_factor_auth_passed"))){
+                $mfaVerified = '1';
+            }
+
             $isAdmin = $this->groupManager->isAdmin($this->userId);
             $userRoot = $this->rootFolder->getUserFolder($this->userId);
 
             try {
                 $node = $userRoot->get($source);
-                $hasAccess = $isAdmin || $node->getOwner()->getUID() === $this->userId;
+                $hasAccess = $isAdmin || ($node->getOwner()->getUID() === $this->userId && $mfaVerified === '1');
```

Comment:
I think this fixes https://git.radicallyopensecurity.com/ros/pen-sunet-mfa-zones/-/issues/2